Resolves #6751 - added kafka consumer assign method interception to p…

Yebemeto opened the pull request:
…roperly trace requests going through 'assigned' topics

<!--
    ⚠️ Please make sure to read this template first, pull requests that don't accord with this template
    maybe closed without notice.
    Texts surrounded by `<` and `>` are meant to be replaced by you, e.g. <framework name>, <issue number>.
    Put an `x` in the `[ ]` to mark the item as CHECKED. `[x]`
-->

<!-- ==== 🐛 Remove this line WHEN AND ONLY WHEN you're fixing a bug, follow the checklist 👇 ====
### Fix <bug description or the bug issue number or bug issue link>
- [ ] Add a unit test to verify that the fix works.
- [ ] Explain briefly why the bug exists and how to fix it.
     ==== 🐛 Remove this line WHEN AND ONLY WHEN you're fixing a bug, follow the checklist 👆 ==== -->

<!-- ==== 🔌 Remove this line WHEN AND ONLY WHEN you're adding a new plugin, follow the checklist 👇 ====
### Add an agent plugin to support <framework name>
- [ ] Add a test case for the new plugin, refer to [the doc](https://github.com/apache/skywalking/blob/master/docs/en/guides/Plugin-test.md)
- [ ] Add a component id in [the component-libraries.yml](https://github.com/apache/skywalking/blob/master/oap-server/server-bootstrap/src/main/resources/component-libraries.yml)
- [ ] Add a logo in [the UI repo](https://github.com/apache/skywalking-rocketbot-ui/tree/master/src/views/components/topology/assets)
     ==== 🔌 Remove this line WHEN AND ONLY WHEN you're adding a new plugin, follow the checklist 👆 ==== -->

<!-- ==== 📈 Remove this line WHEN AND ONLY WHEN you're improving the performance, follow the checklist 👇 ====
### Improve the performance of <class or module or ...>
- [ ] Add a benchmark for the improvement, refer to [the existing ones](https://github.com/apache/skywalking/blob/master/apm-commons/apm-datacarrier/src/test/java/org/apache/skywalking/apm/commons/datacarrier/LinkedArrayBenchmark.java)
- [ ] The benchmark result.
```text
<Paste the benchmark results here>
```
- [ ] Links/URLs to the theory proof or discussion articles/blogs. <links/URLs here>
     ==== 📈 Remove this line WHEN AND ONLY WHEN you're improving the performance, follow the checklist 👆 ==== -->

<!-- ==== 🆕 Remove this line WHEN AND ONLY WHEN you're adding a new feature, follow the checklist 👇 ====
### <Feature description>
- [ ] If this is non-trivial feature, paste the links/URLs to the design doc.
- [ ] Update the documentation to include this new feature.
- [ ] Tests(including UT, IT, E2E) are added to verify the new feature.
- [ ] If it's UI related, attach the screenshots below.
     ==== 🆕 Remove this line WHEN AND ONLY WHEN you're adding a new feature, follow the checklist 👆 ==== -->

- [X ] If this pull request closes/resolves/fixes an existing issue, replace the issue number. Closes #6751.
- [ ] Update the [`CHANGES` log](https://github.com/apache/skywalking/blob/master/CHANGES.md).

wu-sheng commented:
I think you should learn about plugin test
1. https://github.com/apache/skywalking/tree/master/test/plugin/scenarios/kafka-scenario should be updated, including test demo codes and expected data file.
2. Changes.md should be updated.

dmsolr commented:
@Yebemeto Hi, cloud you like to collect partition numbers when you collect the topics?

wu-sheng commented:
@Yebemeto Any update?

Yebemeto commented:
@wu-sheng  Hey, I ll take a look very soon, perhaps today.

Yebemeto commented:
@dmsolr In use case i need this for it is not beneficial. You think it should be recorded as part of trace?
This would kind of change how both subscribe and assign traces look in kafka.

dmsolr commented:
> @dmsolr In use case i need this for it is not beneficial. You think it should be recorded as part of trace?
> This would kind of change how both subscribe and assign traces look in kafka.

I got it. Thanks for your explanation.

darcydai commented on apm-sniffer/apm-sdk-plugin/kafka-plugin/src/test/java/org/apache/skywalking/apm/plugin/kafka/AssignMethodInterceptorTest.java:
```diff
@@ -0,0 +1,70 @@
+    };
+
+    @Before
+    public void setUp() {
```
setup

Yebemeto replied:
thanks, done.

zifeihan commented:
Please check CI status, some checkStyle was failed.

Yebemeto commented:
Ok, thanks for your input everybody. I think i have now done everything.
@wu-sheng I made plugin test cases for this and also updated CHANGES.md
@zifeihan i fixed code style

wu-sheng commented:
@darcydai @dmsolr @zifeihan Please review.

zifeihan reviewed (approved):
thanks, LGTM

dmsolr reviewed (approved):
lgtm
